operators/text/string_regex_replace.cc

```diff
@@ -5,7 +5,11 @@
 #include <vector>
 #include <cmath>
 #include <algorithm>
+#ifdef ENABLE_RE2
```

Comment:
I guess the reason why we replace it with std::regex is to save some binary size. 
In that case, if this operator can be fully implemented by std::regex, then we do it by std::regex.
Otherwise, the operator has to depend on google::re2. so we can remove this kind of ifdef/endif